### ISSUE TITLE
Cannot export performances graphs to image after upgrading to 22.04

### ISSUE THREAD

**Issue by snk33** [centreon-archive/centreon-archived#11770]
# BUG REPORT INFORMATION

### Prerequisites

> The opened issue, must be code related. GitHub is not meant for support. Feel free to check the CONTRIBUTING section for more details.

***Versions***

```
centreon-22.04.4-2.el7.centos.noarch
centreon-auto-discovery-server-22.04.1-2.el7.centos.noarch
centreon-awie-22.04.0-1.el7.centos.noarch
centreon-broker-22.04.1-4.el7.x86_64
centreon-broker-cbd-22.04.1-4.el7.x86_64
centreon-broker-cbmod-22.04.1-4.el7.x86_64
centreon-broker-core-22.04.1-4.el7.x86_64
centreon-broker-debuginfo-21.10.0-6.el7.x86_64
centreon-broker-devel-22.04.1-4.el7.x86_64
centreon-broker-graphite-22.04.1-4.el7.x86_64
centreon-broker-influxdb-22.04.1-4.el7.x86_64
centreon-broker-storage-22.04.1-4.el7.x86_64
centreon-central-22.04.4-2.el7.centos.noarch
centreon-clib-22.04.1-4.el7.x86_64
centreon-common-22.04.4-2.el7.centos.noarch
centreon-connector-22.04.1-4.el7.x86_64
centreon-connector-perl-22.04.1-4.el7.x86_64
centreon-connector-ssh-22.04.1-4.el7.x86_64
centreon-database-22.04.4-2.el7.centos.noarch
centreon-engine-22.04.1-4.el7.x86_64
centreon-engine-daemon-22.04.1-4.el7.x86_64
centreon-engine-extcommands-22.04.1-4.el7.x86_64
centreon-gorgone-22.04.0-9.el7.centos.noarch
centreon-gorgone-centreon-config-22.04.0-9.el7.centos.noarch
centreon-license-manager-22.04.0-1.el7.centos.noarch
centreon-license-manager-common-22.04.0-1.el7.centos.noarch
centreon-perl-libs-22.04.4-2.el7.centos.noarch
centreon-poller-22.04.4-2.el7.centos.noarch
centreon-pp-manager-22.04.0-2.el7.centos.noarch
centreon-release-22.04-3.el7.centos.noarch
centreon-trap-22.04.4-2.el7.centos.noarch
centreon-web-22.04.4-2.el7.centos.noarch
centreon-widget-engine-status-22.04.0-5.el7.centos.noarch
centreon-widget-global-health-22.04.0-5.el7.centos.noarch
centreon-widget-graph-monitoring-22.04.0-5.el7.centos.noarch
centreon-widget-grid-map-22.04.0-5.el7.centos.noarch
centreon-widget-hostgroup-monitoring-22.04.0-5.el7.centos.noarch
centreon-widget-host-monitoring-22.04.0-5.el7.centos.noarch
centreon-widget-httploader-22.04.0-5.el7.centos.noarch
centreon-widget-live-top10-cpu-usage-22.04.0-5.el7.centos.noarch
centreon-widget-live-top10-memory-usage-22.04.0-5.el7.centos.noarch
centreon-widget-servicegroup-monitoring-22.04.0-5.el7.centos.noarch
centreon-widget-service-monitoring-22.04.0-5.el7.centos.noarch
centreon-widget-single-metric-22.04.0-5.el7.centos.noarch
centreon-widget-tactical-overview-22.04.0-5.el7.centos.noarch
```

***Operating System***

CentOS 7

***Browser used***

- [ ] Google Chrome
- [x] Firefox
- [ ] Internet Explorer IE11
- [x] Safari

Version: --

***Additional environment details (AWS, VirtualBox, physical, etc.):***

Hosted on a VM on a VMware Private Cloud

### Description

After upgrading from 21.10 to 22.04, we cannot export performance graph anymore.

### Steps to Reproduce

1. I logged in Centreon
2. I reached the Performances -> Graphs menu
3. I search for some graph4. 
4. After setting the time range, I try to download the graph clicking the button

### Describe the received result

The page is empty, using the web inspector we can see the /centreon/include/views/graphs/generateGraphs/generateImage.php request returns a 500 error code.

### Describe the expected result

A PNG file must be downloaded and the page should remain opened with the graph.

### Logs

**PHP error logs**

```
[15-Sep-2022 16:46:49 Europe/Paris] PHP Fatal error:  Uncaught Exception: Username and token query strings must be set. in /usr/share/centreon/www/include/views/graphs/generateGraphs/generateImage.php:99
Stack trace:
#0 {main}
  thrown in /usr/share/centreon/www/include/views/graphs/generateGraphs/generateImage.php on line 99
[15-Sep-2022 16:47:02 Europe/Paris] PHP Fatal error:  Uncaught Error: Object of class Centreon\Domain\Menu\Model\Page could not be converted to string in /usr/share/centreon/www/class/centreonDBStatement.class.php:129
Stack trace:
#0 /usr/share/centreon/www/class/centreonDBStatement.class.php(129): PDOStatement->execute()
#1 /usr/share/centreon/www/class/centreonDB.class.php(231): CentreonDBStatement->execute()
#2 /usr/share/centreon/www/index.php(220): CentreonDB->execute()
#3 /usr/share/centreon/www/index.php(55): manageRedirection()
#4 {main}
  thrown in /usr/share/centreon/www/class/centreonDBStatement.class.php on line 129
```

### Additional relevant information (e.g. frequency, ...)

Downloading a graph from Resources status page is working as expected (but the render is not as good as the Performances menu).

Here is the web request exported as a curl command (showing that the "token" GET parameter is indeed missing) :
```
curl 'https://xxx/centreon/include/views/graphs/generateGraphs/generateImage.php?chartId=1747_23440&start=1662028721&end=1663238321' -X 'GET' -H 'Cookie: PHPSESSID=xxx; _ga=xxx; _gid=xxx; _gcl_au=xxx' -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' -H 'Accept-Encoding: gzip, deflate, br' -H 'Host: XXX' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.6.1 Safari/605.1.15' -H 'Accept-Language: fr-FR,fr;q=0.9' -H 'Referer: https://xxxx/centreon/main.get.php?p=20401' -H 'Connection: keep-alive'
```

**Comment by ghost** [comment]
Hi, just to offer more information:

Since the following  `else`  code was added is not possible to use `/usr/share/centreon/www/include/views/graphs/generateGraphs/generateImage.php`. Every time when this is called from GUI it logged in this error:

```php
 97     }
 98 } else {
 99     throw new \Exception('Username and token query strings must be set.');
100 }
```
The error log:

```sh
[19-Sep-2022` 17:55:08 Europe/Madrid] PHP Fatal error:  Uncaught Exception: Username and token query strings must be set. in /usr/share/centreon/www/include/views/graphs/generateGraphs/generateImage.php:99
Stack trace:
#0 {main}
  thrown in /usr/share/centreon/www/include/views/graphs/generateGraphs/generateImage.php on line 99
  ```

It's like in this point of the code, the vars `$userName` and/or `$token` are empty. 

If you comment this `else` it starts to works properly:

```php
 97     }
 98 } /*else {
 99     throw new \Exception('Username and token query strings must be set.');
100 }*/
```


Kind regards
David C.

**Comment by ponchoh** [comment]
I believe we have an internal case for this and added this to the case. Thank you.

**Comment by ghost** [comment]
Hello @snk33 :

It has been fixed in centreon-web-22.04.6
